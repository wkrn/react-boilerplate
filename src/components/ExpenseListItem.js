import moment from 'moment';
import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link to={`/edit/${id}`}>
        <div className="list-item">
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__subtitle">{moment(createdAt).format("YYYY-MM-DD")}</span>
            <h3 className="list-item__data">{numeral(amount / 100).format('$0.0.00')}</h3>
        </div>

    </Link>
)

export default ExpenseListItem