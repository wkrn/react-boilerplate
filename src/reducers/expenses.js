const expensesReducerDefaultState = [];
export default (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [...state, action.expense]
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => {
                return id !== action.id
            })
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    console.log(...action.updates)
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense
                }
            })
        case 'SET_EXPENSES':
            return action.expenses;
        default:
            return state
    }
}
