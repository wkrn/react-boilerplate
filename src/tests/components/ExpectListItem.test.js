import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem'


test('should render ExpenseListItem correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(toJSON(wrapper)).toMatchSnapshot()

})