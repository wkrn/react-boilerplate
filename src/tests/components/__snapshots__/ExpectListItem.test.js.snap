// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<Link
  replace={false}
  to="/edit/1"
>
  <div
    className="list-item"
  >
    <h3
      className="list-item__title"
    >
      Gum
    </h3>
    <span
      className="list-item__subtitle"
    >
      1970-01-01
    </span>
    <h3
      className="list-item__data"
    >
      $2.0
    </h3>
  </div>
</Link>
`;
